cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseMetricsTable.java

```diff
@@ -115,7 +124,7 @@ public void rejectedExecution(Runnable r, ThreadPoolExecutor executor) {
         }
       };
 
-      int maxScanThread = spec.getIntProperty(Constants.Metrics.METRICS_HBASE_MAX_SCAN_THREADS, 96);
+      int maxScanThread = cConf.getInt(Constants.Metrics.METRICS_HBASE_MAX_SCAN_THREADS, 96);
```

Comment:
Is this property in cdap-default.xml? In that case you don't need a default here.